src/components/field/svg/FieldObstacles.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from "react";
+import DocumentManagerContext from "../../../document/DocumentManager";
+import { observer } from "mobx-react";
+import { Circle } from "@mui/icons-material";
+import { Box } from "@mui/material";
+import * as d3 from "d3";
+import { ICircularObstacleStore } from "../../../document/CircularObstacleStore";
+
+type Props = { obstacle: ICircularObstacleStore; index: number };
+
+type State = {};
+const STROKE = 0.1;
+const DOT = 0.1;
+
+class FieldGrid extends Component<Props, State> {
+  static contextType = DocumentManagerContext;
+  context!: React.ContextType<typeof DocumentManagerContext>;
+  state = {};
+  rootRef: React.RefObject<SVGElement> = React.createRef<SVGElement>();
+
+  appendIndexID(id: string): string {
+    return `${id}${this.props.index}`;
+  }
+
+  dragPointTranslate(event: any) {
+    let pointerPos = { x: 0, y: 0 };
+    pointerPos.x = event.x;
+    pointerPos.y = event.y;
+
+    this.props.obstacle.setX(pointerPos.x);
```

Comment:
This snaps the obstacle center to the pointer, which does not work well on drag targets the size of the obstacle.